addons/hr/models/hr_employee.py

```diff
@@ -186,6 +186,14 @@ def _compute_work_permit_name(self):
             permit_no = '_' + employee.permit_no if employee.permit_no else ''
             employee.work_permit_name = "%swork_permit%s" % (name, permit_no)
 
+    @api.onchange('resource_calendar_id')
+    def _onchange_resource_calendar_id(self):
+        if not self.resource_calendar_id:
+            if hasattr(self, 'contract_id') and self.contract_id and self.contract_id.resource_calendar_id:
+                self.resource_calendar_id = self.contract_id.resource_calendar_id
+            else:
+                self.resource_calendar_id = self.company_id.resource_calendar_id
```

Comment:
to do this cleanly (without relying on another module), the contract_id part should be done in an override, like here:

```suggestion
            self.resource_calendar_id = self.company_id.resource_calendar_id
```

and then in module `hr_contract`, this `onchange` could be overridden by something like:

```py
    def _onchange_resource_calendar_id(self):
        if not self.resource_calendar_id:
            if self.contract_id.resource_calendar_id:
                self.resource_calendar_id = self.contract_id.resource_calendar_id
        super()._onchange_resource_calendar_id()
```